redis_websocket_api/server.py

```diff
@@ -49,13 +49,25 @@ async def websocket_handler(self, websocket, path):
 
         logger.info("Client %s connected", websocket.remote_address)
         handler = await self.handler_class.create(
-            self.redis, websocket, read_timeout=self.read_timeout,
+            self.redis,
+            websocket,
+            read_timeout=self.read_timeout,
         )
         self.handlers[websocket.remote_address] = handler
+        handler_listen_task = asyncio.create_task(
+            asyncio.wait_for(handler.listen(), self.keep_alive_timeout)
+        )
         try:
-            await asyncio.wait_for(handler.listen(), self.keep_alive_timeout)
+            await asyncio.wait(
```

Comment:
With the new variant, exceptions from `asyncio.wait_for(handler.listen(), self.keep_alive_timeout)` (`TimeOutError` and other exceptions) are silently ignored. Before, they were raised. Is that on purpose?

Reply:
Digging down the rabbit hole I found a few more things to improve... `handler_listen_task` is awaited now, also `handler.close` makes sure waiting for a message in `_queue_reader` is cancelled (meaning `handler_listen_task` is typically done once the `finally` below is reached).

Reply:
Looks good to me now.
The only thing I'm not really sure about is if it can happen that the `asyncio.create_task(websocket.wait_closed())` task can be garbage-collected before it finishes (and if that would harm anyone if it happens). I assume that `asyncio.wait` keeps a reference to all tasks but as soon as it returns, there is no reference to that task anymore but the connection might still be open at that point, so the task might still be running. Maybe it would be more clean to await it (maybe by using the return values from `asyncio.wait`).

Reply:
`wait_closed` is just a synchronization helper, that's why I didn't really care about how it's cleaned up after the `wait`. But I guess it's better to avoid loose tasks all together, so I'll give it a name